### PR TITLE
#179 Fix/association property edit

### PR DESCRIPTION
**Pull request by kshradhan**
| Q                       | A                                |
| ----------------------- | -------------------------------- |
| Fixed issues?           | Fixes: #179  |
| Tests added + passed?   | Yes                              |
| Changeset added?        | No |
| Documentation PR link   |                                  |
| Any dependency changes? | No                               |

Added a check to delete otherProperty from associations only when it is already populated and not when it is first created

### CODE REVIEW

File: packages/legend-studio/src/models/metamodels/pure/model/packageableElements/domain/Association.ts [hunk 1]
```diff
@@ -139,13 +139,14 @@ export class Association
...
       `Association property '${property.name}' must be of type 'class'`,
     );
-    assertTrue(
-      deleteEntry(
-        otherPropertyAssociatedClass.propertiesFromAssociations,
-        otherProperty,
-      ),
-      `Can't find property '${otherProperty.name}' from association '${this.path}' in associated class '${otherPropertyAssociatedClass.path}'`,
-    );
+    otherPropertyAssociatedClass.propertiesFromAssociations.length &&
```
Comment (akphi) [inline packages/legend-studio/src/models/metamodels/pure/model/packageableElements/domain/Association.ts:142]: can we use a normal `if` here?

**Review by akphi — CHANGES_REQUESTED** [review]

**Comment by akphi** [comment]
@kshradhan you would need to sign the CLA, also, make sure to add changeset to your change.

File: packages/legend-studio/src/models/metamodels/pure/model/packageableElements/domain/Association.ts (hunk 1 above)
Comment (akphi) [reply packages/legend-studio/src/models/metamodels/pure/model/packageableElements/domain/Association.ts:142]: Also my concern here why is the `propertiesFromAssociations` of the other class is empty? Did we mess up building the graph at some point?

**Review by MauricioUyaguari — COMMENTED** [review]
Please add an association editor test here. Follow the UMLEditortest. Make it simple.

**Comment by kshradhan** [comment]
> Please add an association editor test here. Follow the UMLEditortest. Make it simple.

Done

File: .changeset/healthy-kings-give.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@finos/legend-studio': patch
+---
+
+Add test for Association Viewer
```
Comment (akphi) [inline .changeset/healthy-kings-give.md:5]: You don't need a message when you add test like this. You can delete this changeset as the other one should already suffice

File: packages/legend-studio/src/components/editor/edit-panel/uml-editor/__tests__/UMLEditor.test.tsx [hunk 3]
```diff
@@ -186,4 +186,52 @@ test(integrationTest('Enumeration View'), async () => {
...
+  fireEvent.click(navigateToPropertyButton);
+  await waitFor(() => getByText(associationEditor, 'property'));
+  const subPropertyPanel = getByTestId(associationEditor, CORE_TEST_ID.PANEL);
+  getByDisplayValue(subPropertyPanel, 'association tag');
+  fireEvent.click(getByText(subPropertyPanel, 'Stereotypes'));
+  await waitFor(() => getByText(subPropertyPanel, 'stereotype1'));
+  const deleteSubPanelButton = queryAllByRole(subPropertyPanel, 'button')[0];
+  fireEvent.click(deleteSubPanelButton);
+});
 // Unable to find an element with the text: temporal. This could be because the text is broken up by multiple elements. In this case, you can provide a function for your text matcher to make your matcher more flexible.
```
Comment (akphi) [inline packages/legend-studio/src/components/editor/edit-panel/uml-editor/__tests__/UMLEditor.test.tsx:237]: :) can you help us delete this out-of-context comment? Thanks!

Comment (akphi) [reply packages/legend-studio/src/components/editor/edit-panel/uml-editor/__tests__/UMLEditor.test.tsx:237]: Ah, nvm, I took care of it

File: packages/legend-studio/src/models/metamodels/pure/model/packageableElements/domain/Association.ts [hunk 4]
```diff
@@ -139,13 +139,15 @@ export class Association
...
       `Association property '${property.name}' must be of type 'class'`,
     );
-    assertTrue(
-      deleteEntry(
-        otherPropertyAssociatedClass.propertiesFromAssociations,
-        otherProperty,
-      ),
-      `Can't find property '${otherProperty.name}' from association '${this.path}' in associated class '${otherPropertyAssociatedClass.path}'`,
-    );
+    if (otherPropertyAssociatedClass.propertiesFromAssociations.length) {
```
Comment (akphi) [inline packages/legend-studio/src/models/metamodels/pure/model/packageableElements/domain/Association.ts:142]: Like I mentioned before, I get this error, but what you're doing here is handling the code smell, you haven't addressed the actual problem, which is the fact that we messed up graph building somewhere.

**Comment by kshradhan** [comment]
Closing this due to misconfigured histories because of wrong email address

**Pull request closed without merging**